mne/minimum_norm/DeFleCT.py

```diff
@@ -0,0 +1,463 @@
+    OH July 2015
+    """
+
+    # check if number of labels and number of SVD components are compatible
+    len_labels = len(labels)
+    len_svd = len(n_svd_comp)    
+    if len_svd > 1:    # if numbers supposed to exist for every label
+        if len_svd != len_labels:
+            raise ValueError("\nNumber of labels and SVD components do not \
+                                match: %d vs %d" % (len_labels, len_svd))
+    else:    # if only one number to be applied to all labels, except first
+        if len_labels > 1: # if only one label: keep n_svd_comp as is
+            # as specified for non-target labels
+            n_svd_comp = np.tile(n_svd_comp[0], len_labels).tolist()
+            n_svd_comp[0] = 1     # one component for first label
+
+    label_lfd_summary = np.array([])  # sub-leadfield summaries
+    fwd_idx_all = []  #  source space indices in labels
+
+    for [lli,ll] in enumerate(labels):
+        logger.info(ll)
+        if ll.hemi == 'rh':
+            # for RH labels, add number of LH vertices
+            offset = forward['src'][0]['vertno'].shape[0]
+            # remember whether we are in the LH or RH
+            this_hemi = 1
+        elif ll.hemi == 'lh':
+            offset = 0
+            this_hemi = 0
+
+        # get vertices on cortical surface inside label
+        idx = np.intersect1d(ll.vertices, forward['src'][this_hemi]['vertno'])
+
+        # get vertices in source space inside label
+        fwd_idx = np.searchsorted(forward['src'][this_hemi]['vertno'], idx)
+        fwd_idx_all.append(fwd_idx)
+
+        # get sub-leadfield matrix for label vertices
+        sub_leadfield = leadfield[:, fwd_idx + offset]
+
+        # compute summary data for labels
+        if mode == 'sum':  # sum across forward solutions in label
+            logger.info("Computing sums within labels")
+            this_label_lfd_summary = np.array(sub_leadfield.sum(axis=1))        
+        elif mode == 'mean':
+            logger.info("Computing means within labels")
+            this_label_lfd_summary = np.array(sub_leadfield.mean(axis=1))        
+        elif mode == 'svd':  # takes svd of forward solutions in label
+            ch_names = forward['info']['ch_names']
+
+            this_label_lfd_summary = label_svd(sub_leadfield, n_svd_comp[lli],
+                                                                      ch_names)
+
+        # initialise or append to existing list
+        if lli == 0:
+            label_lfd_summary = this_label_lfd_summary
+        else:            
+            label_lfd_summary = np.append(label_lfd_summary,
+                                           this_label_lfd_summary, 1)
+
+    return label_lfd_summary
+
+# Done
+
+
+def DeFleCT_make_estimator(forward, noise_cov, labels, lambda2_cov=3/10.,
+                           lambda2_S=1/9., pick_meg=True, pick_eeg=False,
+                           mode='svd', n_svd_comp=1, verbose=None):
```

Comment:
I don't have a strong opinion on this - but I think you are right, DeFleCT isn't a minimum norm method. But that's also where my "psf_ctf" functions are. I have not changed this yet.